### PR TITLE
DM-46061: Add Metrics to calibrate task metadata

### PR DESCRIPTION
**Pull request by jrmullaney**

### CODE REVIEW

File: pipelines/calexpMetrics.yaml [hunk 1]
```diff
@@ -6,3 +6,32 @@ tasks:
     config:
       atools.calexpSummaryMetrics: CalexpSummaryMetrics
       python: from lsst.analysis.tools.atools import *
+  analyzeCalibrateMetadata:
+    class: lsst.analysis.tools.tasks.MetadataAnalysisTask
+    config:
+      connections.taskName: calibrate
+      connections.outputName: calibrate_metadata
+      metadataDimensions: [instrument, visit, detector]
```
Comment (isullivan) [inline pipelines/calexpMetrics.yaml:14]: Do these need to be `["instrument", "visit", "detector"]`?

File: python/lsst/analysis/tools/tasks/metadataAnalysis.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+class MetadataAnalysisTask(AnalysisPipelineTask):
+    ConfigClass = MetadataAnalysisConfig
+    _DefaultName = "metadataAnalysis"
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        dataId = butlerQC.quantum.dataId
+        inputs = butlerQC.get(inputRefs)
+        plotInfo = self.parsePlotInfo(inputs, dataId)
+        taskName = inputRefs.data.datasetType.name
+        taskName = taskName[: taskName.find("_")]
```
Comment (isullivan) [inline python/lsst/analysis/tools/tasks/metadataAnalysis.py:86]: Not terribly important, but you might as well move the `taskName` definition inside the `if not metadata:` block below.

**Review by isullivan — APPROVED** [review]
Looks good. We should try to replace both `metadataExposureDetectorAnalysis.py` and `diffimTaskDetectorVisitMetricsAnalysis.py` with calls to your new generic class. Those can be done on new tickets, though.

File: python/lsst/analysis/tools/tasks/metadataAnalysis.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+class MetadataAnalysisConnections(
+    AnalysisBaseConnections,
+    dimensions={"instrument", "exposure", "detector"},
+    defaultTemplates={"taskName": "isr"},
+):
+    data = connectionTypes.Input(
+        doc="Task metadata to load.",
+        name="{taskName}_metadata",
+        storageClass="TaskMetadata",
+        deferLoad=True,
```
[the remark this answers is not in the document]
Comment (jrmullaney) [reply python/lsst/analysis/tools/tasks/metadataAnalysis.py:40]: `deferLoad` is needed for the `plotInfo = self.parsePlotInfo(inputs, dataId)` in `def runQuantum(...` below. Without it, `tableName = inputs[connectionName].ref.datasetType.name` in `parsePlotInfo` returns the attribute error `'TaskMetadata' object has no attribute 'ref'`.

**Pull request merged**